kuma/static/js/newsletter.js

```diff
@@ -0,0 +1,221 @@
+        } else if (isArticle && isHidden) {
+            newsletterHide();
+        }
+    }
+
+    function newsletter() {
+
+        var newsletterEmailInput = doc.getElementById('newsletterEmailInput');
+        var newsletterPrivacy = doc.getElementById('newsletterPrivacy');
+
+        // handle errors
+        var errorArray = [];
+        var newsletterErrors = doc.getElementById('newsletterErrors');
+        function newsletterError(event) {
+            var errorList = doc.createElement('ul');
+            errorList.className ='errorlist';
+
+            if(errorArray.length) {
+                for (var i = 0; i < errorArray.length; i++) {
+                    var item = doc.createElement('li');
+                    item.appendChild(doc.createTextNode(errorArray[i]));
+                    errorList.appendChild(item);
+                }
+            } else {
+                // no error messages, forward to server for better troubleshooting
+                newsletterForm.setAttribute('data-skip-xhr', true);
+                newsletterForm.submit();
+            }
+
+            newsletterErrors.appendChild(errorList);
+            newsletterErrors.style.display = 'block';
+
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'error'
+            });
+        }
+
+        // show sucess message
+        function newsletterThanks() {
+            var thanks = doc.getElementById('newsletterThanks');
+
+            // show thanks message
+            thanks.style.display = 'block';
+
+            // track success
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'complete'
+            });
+
+            // don't show signup form on article pages anymore
+            newsletterSaveHide();
+        }
+
+        // XHR subscribe; handle errors; display thanks message on success.
+        function newsletterSubscribe(event) {
+            var skipXHR = newsletterForm.getAttribute('data-skip-xhr');
+            if (skipXHR) {
+                // track this
+                mdn.analytics.trackEvent({
+                    'category': 'newsletter',
+                    'action': 'progression',
+                    'label': 'error-forward'
+                });
+
+                return true;
+            }
+            event.preventDefault();
+            event.stopPropagation();
+
+            // track this submission
+            mdn.analytics.trackEvent({
+                'category': 'newsletter',
+                'action': 'progression',
+                'label': 'submission'
+            });
+
+            // new submission, clear old errors
+            errorArray = [];
+            newsletterErrors.style.display = 'none';
+            while (newsletterErrors.firstChild) newsletterErrors.removeChild(newsletterErrors.firstChild);
```

Comment:
MDN uses the same codding styles as bedrock in theory. In addition to changing it here I'll submit a PR to https://github.com/mozilla/basket-example/blob/master/basket-client.js which is where I got the start of this file from. That should also match bedrock styles.